tests/data/data_manager/test_foldermap.py

```diff
@@ -0,0 +1,195 @@
+                [
+                    (
+                        {"molname": "O2", "basis": "STO-3G", "bondlength": "0.5"},
+                        "qchem/O2/STO-3G/0.5.h5",
+                    ),
+                    (
+                        {"molname": "O2", "basis": "STO-3G", "bondlength": "0.6"},
+                        "qchem/O2/STO-3G/0.6.h5",
+                    ),
+                ],
+            ),
+        ],
+    )
+    def test_find(self, foldermap, kwds, expect):  # pylint: disable=redefined-outer-name
+        """Test that the ``find()`` method returns the expected results
+        for a range of arguments."""
+
+        assert set(foldermap.find("qchem", **kwds)) == set(
+            (Description(desc), DataPath(path)) for desc, path in expect
+        )
+
+    @pytest.mark.parametrize(
+        "init, key, expect",
+        [
+            (FOLDERMAP, "qchem", FolderMapView(FOLDERMAP["qchem"])),
+            (FOLDERMAP["qchem"], "H2", FolderMapView(FOLDERMAP["qchem"]["H2"])),
+            (FOLDERMAP["qchem"]["O2"]["STO-3G"], "0.5", DataPath("qchem/O2/STO-3G/0.5.h5")),
+            (FOLDERMAP["qchem"]["O2"], DEFAULT, FolderMapView(FOLDERMAP["qchem"]["O2"]["STO-3G"])),
+            (FOLDERMAP["qchem"]["O2"]["STO-3G"], "0.6", DataPath("qchem/O2/STO-3G/0.6.h5")),
+            (FOLDERMAP["qchem"]["O2"]["STO-3G"], DEFAULT, DataPath("qchem/O2/STO-3G/0.5.h5")),
+        ],
+    )
+    def test_gettitem(self, init, key, expect):
+        """Test that ``getitem`` returns the expected values, including
+        for default values and nested foldermaps."""
+        assert FolderMapView(init)[key] == expect
+
+    @pytest.mark.parametrize("key", ["__default"])
+    @pytest.mark.parametrize(
+        "init",
+        [
+            FOLDERMAP,
+            FOLDERMAP["qchem"],
+            FOLDERMAP["qchem"]["O2"],
+        ],
+    )
+    def test_getitem_private_key(self, key, init):
+        """Test that ``getitem`` raises a KeyError if the ``__default``
+        key is passed."""
+
+        with pytest.raises(KeyError):
+            _ = FolderMapView(init)[key]
+
+    @pytest.mark.parametrize(
+        "init",
+        [
+            FOLDERMAP,
+            FOLDERMAP["qchem"],
+        ],
+    )
+    def test_getitem_default_none(self, init):
+        """Test that ``getitem`` raises a ValueError if ``DEFAULT`` is
+        used, but there is no default defined for that level."""
+
+        with pytest.raises(ValueError, match="No default available"):
+            _ = FolderMapView(init)[DEFAULT]
+
+    @pytest.mark.parametrize(
+        "init, keys",
+        [
+            (FOLDERMAP, {"qchem"}),
+            (FOLDERMAP["qchem"], {"O2", "H2"}),
+            (FOLDERMAP["qchem"]["O2"], {"STO-3G", "6-31G"}),
+            (FOLDERMAP["qchem"]["O2"]["STO-3G"], {"0.5", "0.6"}),
+        ],
+    )
+    def test_keys(self, init, keys):
+        """Test that the ``keys()`` method returns only publicly visible keys."""
```

Comment:
do you expect users to call `keys()`? personally, when I want the keys of a dict, I usually just do `set(the_dict)` because `set()` already gives only the keys (same for any iterable too)